### PR TITLE
feat: Comprehensive Testing Infrastructure

### PR DESCRIPTION
**Pull request by jamcalli**
## Overview

Major internal improvement release focused on test coverage, code organization, and stability. Adds 760+ unit and integration tests, refactors core modules into cleaner architecture, and fixes several edge cases in genre handling and webhook processing.

### Comprehensive Test Suite
- **760+ tests added**: Unit and integration tests across all major subsystems
- **MSW integration**: Mock Service Worker for reliable API testing without network calls
- **Isolated test databases**: Per-process SQLite databases with automatic cleanup
- **Mock infrastructure**: Reusable mocks for logger, abort signals, fetch, and streams
- **Type-safe test suite**: Full TypeScript support with dedicated tsconfig

### Critical Bug Fixes
- **Genre normalization**: Fixed handling of JSON strings, arrays, and null values in genre data
- **Webhook cache stability**: Enhanced cache key generation to prevent duplicate webhook processing
- **Discord mention privacy**: Users who opt out of Discord notifications are no longer mentioned in public channels
- **Test database isolation**: Fixed sequence reset to ensure deterministic test IDs

#### Plex Module Refactoring
- Split monolithic `plex.ts` into focused modules:
  - `api-client.ts`: Core API communication (ping, avatar fetching)
  - `friends-api.ts`: Friend management and GraphQL operations
  - `watchlist-api.ts`: Watchlist fetching and pagination
  - `watchlist-fetcher.ts`: High-level watchlist orchestration
  - `helpers.ts`: Shared utilities and rate limit handling
  - `processors/`: Batch and single-item processing logic
- Improved maintainability and testability of Plex integration

#### Webhook Queue Refactoring
- Split `webhookQueue.ts` into modular subfolder:
  - `queue-state.ts`: Core queue state management
  - `queue-processor.ts`: Webhook processing and notification dispatch
  - `episode-checker.ts`: Episode recency validation
  - `upgrade-tracker.ts`: Upgrade detection and tracking
  - `pending-webhook.ts`: Pending webhook storage and retrieval
- Better separation of concerns and easier testing

#### Notification System Refactoring
- Split notification processor into focused sub-modules:
  - `notification-builder.ts`: Notification object construction
  - `notification-dispatcher.ts`: Orchestration and user matching
  - `notification-sender.ts`: Delivery via Discord, Apprise, Tautulli
  - `url-parser.ts`: Webhook URL parsing and validation
  - `webhook-validator.ts`: Webhook deduplication and validation
- Cleaner architecture with distinct responsibilities

### Test Infrastructure
- **MSW handlers**: Comprehensive mocks for Plex, Sonarr, Radarr, and Discord APIs
- **Seed data**: Realistic test fixtures for users, instances, configs, and watchlist items
- **Database helpers**: Utilities for test database initialization, reset, and cleanup
- **Mock implementations**: Logger, fetch, abort signals, and stream handlers
- **Global setup**: Automatic test database cleanup and MSW server lifecycle

### Utility Enhancements
- **Genre handling**: New `parseGenres()` utility matching `parseGuids()` pattern
- **GUID parsing**: Enhanced deduplication and type safety
- **Date serialization**: Improved handling of empty objects and null values
- **Quality profile parsing**: Safe integer validation for all numeric IDs

### Documentation Improvements
- **Comprehensive test docs**: README in test/mocks explaining mock patterns
- **Inline documentation**: Enhanced JSDoc across refactored modules
- **Type annotations**: Explicit types for better IDE support and maintainability

## Migration & Compatibility

**No breaking changes** - All changes are internal. No database migrations or configuration updates required.

**Node version support** - Updated to Node.js v22.21.0 LTS for improved performance and security.

### Core Systems (100% coverage)
- Authentication & authorization
- Content routing & rule building
- Database operations
- Logger with error serialization
- Quality profile parsing
- Session management
- URL utilities

### Plex Integration (100% coverage)
- API client operations
- Friends management
- Watchlist fetching & pagination
- Rate limiting
- Batch processing
- Single-item processing
- RSS feed parsing

### Notification System (100% coverage)
- Notification building & dispatching
- Multi-channel delivery (Discord, Apprise, Tautulli)
- URL parsing
- Webhook validation & deduplication

### Webhook Queue (100% coverage)
- Queue state management
- Episode checking & filtering
- Upgrade tracking
- Pending webhook storage

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [x] Code refactoring
- [ ] Documentation update
- [x] Dependency update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes work with existing functionality

## Summary by CodeRabbit

* **Chores**
  * Updated Node/Docker tooling and CI to newer runtimes and now run tests; bumped package version and added ignore/packaging rules.

* **New Features**
  * Reworked notification/webhook surface into consolidated APIs with queuing, processing, and public-content dispatch.
  * New Plex integration utilities for watchlist and RSS retrieval.
  * SPA asset path handling now respects configurable basePath for reverse-proxy setups.

* **Bug Fixes**
  * Guarded webhook timeout clearing to avoid runtime errors.

* **Tests**
  * Extensive unit/integration test suites, test helpers, MSW setup, and test scripts added.

### CODE REVIEW

File: src/utils/plex/friends-api.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      }
+
+      if (json.data?.allFriendsV2) {
+        const friends = json.data.allFriendsV2.map(
+          (friend: { user: { id: string; username: string } }) =>
+            [
+              { watchlistId: friend.user.id, username: friend.user.username },
+              token,
+            ] as [Friend, string],
+        )
```
[the remark this answers is not in the document]
Comment (jamcalli) [reply src/utils/plex/friends-api.ts:85]: While this is not exactly typesafe, the id is actually provided by the database and not returned from the graphql. 

This would break current usage.
